train.py

```diff
@@ -3,340 +3,388 @@
+                       exclude_gpu_ids=args.excude_gpu_list)
+    else:
+        os.environ["CUDA_VISIBLE_DEVICES"] = args.gpu_ids_to_use.replace(" ", ",")
+
+    from utils.dataset_loading_hub import load_dataset
+
+    from models.wresnet import WideResNet
+    import torch
+    import torch.nn as nn
+    import torch.optim as optim
+    from torchvision.utils import save_image
+    import torch.backends.cudnn as cudnn
+    from torch.optim.lr_scheduler import CosineAnnealingLR, MultiStepLR
+    from utils.metric_tracking import MetricTracker, compute_accuracy
+
+    ######################################################################################################### Admin
+    saved_models_filepath, logs_filepath, images_filepath = build_experiment_folder(
+        experiment_name=args.experiment_name, log_path=args.logs_path
+    )
+
+    ######################################################################################################### Data
+
+    (
+        train_set_loader,
+        val_set_loader,
+        test_set_loader,
+        train_set,
+        val_set,
+        test_set,
+        data_shape,
+        num_classes,
+    ) = load_dataset(
+        args.dataset_name,
+        args.data_filepath,
+        batch_size=model_args.batch_size,
+        test_batch_size=model_args.eval_batch_size,
+        num_workers=args.num_workers,
+        download=True,
+        val_set_percentage=0.1,
+    )
+
+    ######################################################################################################### Determinism
+    # Seeding can be annoying in pytorch at the moment. Based on my experience, the below means of seeding
+    # allows for deterministic experimentation.
+    torch.manual_seed(args.seed)
+    np.random.seed(args.seed)  # set seed
+    random.seed(args.seed)
+    device = (
+        torch.cuda.current_device()
+        if torch.cuda.is_available() and args.num_gpus_to_use > 0
+        else "cpu"
+    )
+    args.device = device
+    if torch.cuda.is_available():
+        torch.cuda.manual_seed_all(args.seed)
+        torch.backends.cudnn.deterministic = True
+
+    # Always save a snapshot of the current state of the code. I've found this helps immensely if you find that one of your
+    # many experiments was actually quite good but you forgot what you did
+
+    snapshot_filename = "{}/snapshot.tar.gz".format(saved_models_filepath)
+    filetypes_to_include = [".py"]
+    all_files = []
+    for filetype in filetypes_to_include:
+        all_files += glob.glob("**/*.py", recursive=True)
+    with tarfile.open(snapshot_filename, "w:gz") as tar:
+        for file in all_files:
+            tar.add(file)
+
+    ######################################################################################################### Model
+
+    model = get_model(
+        model_type=args.model.type,
+        num_classes=num_classes,
+        dataset_name=args.dataset_name,
```

Comment:
Lets have a comment after here showing how you can import the model class itself and start up a model without going through get model. I think that is one thing that is immediately useful to know for anyone reusing code.

Reply:
Yep. Sounds good.